app/views/shopping_cart/_product.html.erb

```diff
@@ -1,6 +1,8 @@
-<% presenter = ProductPresenter.new(product) %>
+<% presenter = ProductPresenter.new(cart_item.product) %>
 <tr>
   <td><%= image_tag presenter.image_attached, size: 100 %></td>
-  <td><%= product.name %></td>
-  <td><%= product.price %></td>
+  <td><%= cart_item.product.name %></td>
```

Comment:
https://medium.com/@gioch/design-patterns-law-of-demeter-with-rails-49a44a9689fe
Może warto tutaj zastosować delegate? -> Ale to sugestia :)

Reply:
Zastosowałem rozwiązanie też dla order_item, bo w sumie występowała tam identyczna sytuacja. Zdecydowanie wygodniejsze rozwiązanie przy refactoringu. :)